tests/test_theseus_layer.py

```diff
@@ -26,12 +26,12 @@
 
 
 def model(x, b):
-    return (b[..., :1] * x) ** 2 + b[..., 1:]
+    return b[..., :1] * x**2 + b[..., 1:]
 
 
 def model_grad(x, b):
-    g1 = 2 * (b[..., :1] * x) * x
-    g2 = b[..., 1:] * torch.ones_like(x)
+    g1 = x**2
+    g2 = torch.ones_like(x)
```

Comment:
Should the size here be related to `b[..., 1:]`?

Reply:
Thanks for catching this, not sure what I was thinking when I added that `b` factor. The size is correct, because of the following:

- `b` (optim_var) is shape `(batch_size, 2)`
- `x` (data, constant) is shape `(batch_size, num_points)`
- error is shape `(batch_size, num_points)`

We want the jacobian with respect to `b` be of shape `(batch_size, num_points, 2)`, and it's computed below as the stack of `[g1, g2]` on `axis=2`, so we need for `g2` to be of shape `(batch_size, num_points)`, which is the shape of `x`.